dask/dataframe/io/tests/test_parquet.py

```diff
@@ -613,3 +613,15 @@ def test_writing_parquet_with_compression(tmpdir, compression, engine):
     ddf.to_parquet(fn, compression=compression, engine=engine)
     out = dd.read_parquet(fn, engine=engine)
     assert_eq(out, df, check_index=(engine != 'fastparquet'))
+
+
+def test_writing_parquet_with_unknown_kwargs(tmpdir, engine):
+    fn = str(tmpdir)
+
+    df = pd.DataFrame({'x': ['a', 'b', 'c'] * 10,
+                       'y': [1, 2, 3] * 10})
+    ddf = dd.from_pandas(df, npartitions=3)
+
+    ddf.to_parquet(fn,  engine=engine, unknown_key='unknown_value')
```

Comment:
This should error imo. If the user passes in unknown invalid keyword arguments, they shouldn't be silently ignored. What shouldn't fail is passing in valid keyword arguments like `compression` or `use_dictionary` (for pyarrow).

Reply:
This would require `kwargs` content validation using all the arguments of all the functions used inside the write function, which is impractical and I've never seen it done, as the point of `kwargs` is to have arguments that are not explicitly known, because there are used by some other function. The problem here is that we mix functions like `write_table`, which has `kwargs` and the unknown ones are ignored, and `write_metadata`, which doesn't have `kwargs` and, thus, fails if it gets something unknown in `kwargs`. For the latter case we should only extract required arguments from `kwargs` and ignore the rest as they may be used by some other function, which we don't know, if we decide to use `kwargs`. If we want to restrict the number of arguments used for the particular engine implementation, we should remove `kwargs` altogether from the `_write_<engine>` function and keep only the known ones. This will fail as intended on `write` call in `to_parquet` function; see: https://github.com/jsnowacki/dask/blob/f054102e35b550670052ed3a00b8dcbae0d3ecb1/dask/dataframe/io/parquet.py#L626. Note that using just specific arguments in the write function for the engine will create a new place that any change to the engine has to be implemented as well.